source/postprocess/heat_flux_map.cc

```diff
@@ -40,49 +43,262 @@ namespace aspect
+                  double temperature_time_derivative;
+
+                  if (simulator_access.get_timestep_number() > 1)
+                    temperature_time_derivative = (1.0/time_step) *
+                                                  (temperatures[q] *
+                                                   (2*time_step + old_time_step) / (time_step + old_time_step)
+                                                   -
+                                                   old_temperatures[q] *
+                                                   (1 + time_step/old_time_step)
+                                                   +
+                                                   old_old_temperatures[q] *
+                                                   (time_step * time_step) / (old_time_step * (time_step + old_time_step)));
+                  else if (simulator_access.get_timestep_number() == 1)
+                    temperature_time_derivative =
+                      (temperatures[q] - old_temperatures[q]) / time_step;
+                  else
+                    temperature_time_derivative = 0.0;
+
+                  const double JxW = fe_volume_values.JxW(q);
+
+                  const double density_c_P = out.densities[q] * out.specific_heat[q];
+                  const double latent_heat_LHS =heating_out.lhs_latent_heat_terms[q];
+                  const double material_prefactor = density_c_P + latent_heat_LHS;
+
+                  for (unsigned int i = 0; i<dofs_per_cell; ++i)
+                    {
+                      // thermal conduction part
+                      local_vector(i) += -out.thermal_conductivities[q] *
+                                         (fe_volume_values[simulator_access.introspection().extractors.temperature].gradient(i,q)
+                                          * temperature_gradients[q])
+                                         * JxW;
+
+                      // advection part
+                      local_vector(i) += -material_prefactor * (temperature_gradients[q] * velocities[q]) *
+                                         fe_volume_values[simulator_access.introspection().extractors.temperature].value(i,q) *
+                                         JxW;
+
+                      // source terms
+                      local_vector(i) += heating_out.heating_source_terms[q] *
+                                         fe_volume_values[simulator_access.introspection().extractors.temperature].value(i,q) *
+                                         JxW;
+
+                      // time variation terms
+                      local_vector(i) += -material_prefactor * temperature_time_derivative *
+                                         fe_volume_values[simulator_access.introspection().extractors.temperature].value(i,q) *
+                                         JxW;
+
+                      // TODO: Do we need stabilization terms?
+                      // TODO: Consider operator splitting terms
```

Comment:
What do you mean by operator splitting in this context? Since there is no timestepping for this calculation I wouldn't anticipate the need for any splitting.